lib/Doctrine/ORM/Tools/EntityGenerator.php

```diff
@@ -1380,6 +1454,16 @@ protected function generateEntityStubMethod(ClassMetadataInfo $metadata, $type,
             $methodTypeHint =  '\\' . $typeHint . ' ';
         }
 
+        $methodReturnType = null;
+        if ($this->generateMethodsTypeHinting) {
```

Comment:
This condition calls 2 methods. One for type hinting parameters and the other for the return type.

It could not be move in only one method.